components/merkledb/src/views/metadata.rs

```diff
@@ -337,6 +339,33 @@ impl<T: RawAccess> IndexesPool<T> {
         self.set_len(len + 1);
         (metadata, is_phantom)
     }
+
+    pub(super) fn suffixes<K: BinaryKey + Clone>(&self, prefix: &IndexAddress) -> Vec<K> {
+        let prefix = prefix.fully_qualified_name();
+        let mut keys = Vec::new();
+
+        for entry in self.0.iter::<Vec<u8>, Vec<u8>, Vec<u8>>(&prefix) {
+            let key = get_suffix(entry);
+            keys.push(key);
+        }
+
+        keys
+    }
+}
+
+fn get_suffix<K: BinaryKey + Clone>(entry: (Vec<u8>, Vec<u8>)) -> K {
+    // TODO: simplify this
```

Comment:
I'm not sure what this logic supposed to do. As I imagine `suffixes` method, it should work on "group" addresses only (i.e., addresses not containing the `id_in_group` part; the method signature may be changed to `prefix: &str` correspondingly). The method should iterate over keys that start with `${prefix}\0` and remove the corresponding number of bytes (`prefix.len() + 1`) at the beginning of the key.